rust/src/client.rs

```diff
@@ -89,6 +111,119 @@ impl Session {
 struct SessionManager {
     logger: Logger,
     tx: tokio::sync::mpsc::Sender<command::Command>,
+    selector: QueueSelector,
+}
+
+impl SessionManager {
+    async fn send(
+        &self,
+        message: &models::message::MessageImpl,
+        client: Weak<&Client>,
+    ) -> Result<SendMessageResponse, ClientError> {
```

Comment:
SessionManager helps manage sessions: by maintaining heartbeat, and administrative controlling flow...

Specific send steps should go to `Session` and `SessionManager` should directly forward workload to a target session after looking up the route-table: mapping between topic/queue to host addresses.